provider/dpdk_utils.py

```diff
@@ -0,0 +1,205 @@
+import subprocess
+import paramiko
+import re
+
+# Ensure paramiko is installed
+for pip in ['pip3', 'pip']:
+    try:
+        subprocess.check_call([pip, 'install', '--default-timeout=100', 'paramiko'])
+        import paramiko
+        break
+    except ImportError:
+        continue
+else:
+    print("Failed to install paramiko. Please install it manually.")
+
+
+def install_dpdk(params, session):
+    """
+    Install the dpdk related packages.
+
+    :param params: Dictionary with the test parameters
+    :param session: the session of guest or host
+    """
+
+    cmd = 'yum install -y %s' % params.get("env_pkg")
+    session.cmd(cmd, timeout=360, ignore_all_errors=True)
+
+
+def load_vfio_modules(session):
+    """
+    Load vfio and vfio-pci modules.
+
+    :param session: the session of guest or host
+    """
+
+    if session.cmd_status("lsmod | grep -wq vfio"):
+        session.cmd("modprobe -r vfio-pci")
+        print("vfio-pci module removed")
+
+    if session.cmd_status("lsmod | grep -wq vfio_pci"):
+        session.cmd("modprobe -r vfio")
+        print("vfio module removed")
+
+    session.cmd_output("modprobe vfio enable_unsafe_noiommu_mode=Y")
+    print("vfio module loaded")
```

Comment:
Same above.

Reply:
For the vDPA feature, some cases need to be run on the host. therefore, it's preferable for the dpdk_utils.py file to be usable independently, making it unsuitable to use LOG_JOB.